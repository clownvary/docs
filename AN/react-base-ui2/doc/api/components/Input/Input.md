## Input 
---
UI component of Input

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string &#124; number|&ensp;|false|Sets the value of the Input.
defaultValue|string &#124; number|&ensp;|false|Sets the default value of the Input.
type|string|'text'|false|Sets the type of the Input and can be 'text', 'password', 'button' etc.
size|enum:<br>['sm', 'md', 'lg']|'md'|false|Determines the Input size.
preIcon|string|&ensp;|false|The label icon displayed before (on the left side of) the input field.
preText|string|&ensp;|false|The label text displayed before (on the left side of) the input field.
postIcon|string|&ensp;|false|The label icon displayed after (on the right side of) the input field.
postText|string|&ensp;|false|The label text displayed after (on the right side of) the input field.
errored|bool|&ensp;|false|Determines if the input has errors.
icon|bool|false|false|Whether add 'input-group--icon' class for the input wrapper.
disabled|bool|&ensp;|false|Whether the input is disabled.
className|string|&ensp;|false|Customize class name for the input wrapper.
style|object|&ensp;|false|Determines the style of the input wrapper.
onChange|func|identity|false|The callback function that is triggered when input value changes.
PreComponent|func &#124; element|&ensp;|false|The component displayed after (on the right side of) the input field.
PostComponent|func &#124; element|&ensp;|false|The component displayed before (on the left side of) the input field.
inputRef|func|identity|false|Gets the instance of the Input Component.
children|node|&ensp;|false|Child Node