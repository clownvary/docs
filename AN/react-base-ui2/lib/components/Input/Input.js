'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _defineProperty2 = require('babel-runtime/helpers/defineProperty');var _defineProperty3 = _interopRequireDefault(_defineProperty2);var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);var _defineProperties = require('babel-runtime/core-js/object/define-properties');var _defineProperties2 = _interopRequireDefault(_defineProperties);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');










var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _identity = require('lodash/identity');var _identity2 = _interopRequireDefault(_identity);
var _isRegExp = require('lodash/isRegExp');var _isRegExp2 = _interopRequireDefault(_isRegExp);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);
var _omit = require('lodash/omit');var _omit2 = _interopRequireDefault(_omit);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/** Default PropTypes of Input.
                                                                                                                                                                             * @memberof Input
                                                                                                                                                                            */
var InputPropTypes = {
  /** Sets the value of the Input.
                        * @type {string|number}
                       */
  value: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number]),
  /** Sets the default value of the Input.
                                                                             * @type {string|number}
                                                                            */
  defaultValue: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number]),
  /** Sets the type of the Input and can be 'text', 'password', 'button' etc.
                                                                                    * @type {string}
                                                                                   */
  type: _propTypes.string.isRequired,
  /** Determines the Input size.
                                       * @type {Size}
                                      */
  size: (0, _propTypes.oneOf)(['sm', 'md', 'lg']).isRequired,
  /** The label icon displayed before (on the left side of) the input field.
                                                               * @type {string}
                                                              */
  preIcon: _propTypes.string,
  /** The label text displayed before (on the left side of) the input field.
                               * @type {string}
                              */
  preText: _propTypes.string,
  /** The label icon displayed after (on the right side of) the input field.
                               * @type {string}
                              */
  postIcon: _propTypes.string,
  /** The label text displayed after (on the right side of) the input field.
                                * @type {string}
                               */
  postText: _propTypes.string,
  /** Determines if the input has errors.
                                * @type {boolean}
                               */
  errored: _propTypes.bool,
  /** Whether add 'input-group--icon' class for the input wrapper.
                             * @type {boolean}
                            */
  icon: _propTypes.bool, // Represent `icon-input`
  /** Whether the input is disabled.
   * @type {boolean}
  */
  disabled: _propTypes.bool,
  /** Customize class name for the input wrapper.
                              * @type {string}
                             */
  className: _propTypes.string,
  /** Determines the style of the input wrapper.
                                 * @type {object}
                                */
  style: _propTypes.object, // eslint-disable-line
  /** The callback function that is triggered when input value changes.
   * @type {func}
  */
  onChange: _propTypes.func,
  /** The component displayed after (on the right side of) the input field.
                              * @type {func|element}
                             */
  PreComponent: (0, _propTypes.oneOfType)([_propTypes.func, _propTypes.element]),
  /** The component displayed before (on the left side of) the input field.
                                                                                   * @type {func|element}
                                                                                  */
  PostComponent: (0, _propTypes.oneOfType)([_propTypes.func, _propTypes.element]),
  /** Gets the instance of the Input Component.
                                                                                    * @type {func}
                                                                                    * @param { domNode } input
                                                                                   */
  inputRef: _propTypes.func,
  /** Child Node
                              * @type {node}
                             */
  children: _propTypes.node,
  /** The content for aria-label
                              * @type {string}
                             */
  ariaLabel: _propTypes.string };


/** Default Props for Input */
var InputProps = {
  type: 'text',
  size: 'md',
  icon: false,
  onChange: _identity2.default,
  inputRef: _identity2.default,
  ariaLabel: 'Input box' };


/** UI component of Input */var
Input = function (_PureComponent) {(0, _inherits3.default)(Input, _PureComponent);function Input() {var _ref;var _temp, _this, _ret;(0, _classCallCheck3.default)(this, Input);for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Input.__proto__ || (0, _getPrototypeOf2.default)(Input)).call.apply(_ref, [this].concat(args))), _this), _this.



















    setWrappedComponentInstance = function (input) {
      _this.input = input;

      _this.props.inputRef(input);
    }, _this.

    handleBlur = function (e) {var _this$props =
      _this.props,onBlur = _this$props.onBlur,onLeave = _this$props.onLeave;

      if ((0, _isFunction2.default)(onBlur)) {
        onBlur(e);
      }

      if ((0, _isFunction2.default)(onLeave)) {
        onLeave({ value: e.target.value });
      }
    }, _this.



    handleChange = function (e) {var _this$props2 =
      _this.props,disabled = _this$props2.disabled,onChange = _this$props2.onChange,onValueChange = _this$props2.onValueChange;

      if (disabled) {
        return;
      }

      // If you want to access the event properties in an asynchronous way,
      // you should call `event.persist()`` on the event,
      // which will remove the synthetic event from the pool
      // and allow references to the event to be retained by user code.
      e.persist && e.persist();

      onChange(e);

      if ((0, _isFunction2.default)(onValueChange)) {
        onValueChange({ value: e.target.value });
      }
    }, _this.

    handleKeyPress = function (e) {var _this$props3 =
      _this.props,disabled = _this$props3.disabled,formula = _this$props3.formula;

      if (disabled || !(0, _isRegExp2.default)(formula)) {
        return;
      }

      var nativeEvent = e.nativeEvent || e;

      if (nativeEvent.ctrlKey || nativeEvent.altKey || nativeEvent.metaKey) {return;}

      /*
                                                                                        nativeEvent.char  - IE11
                                                                                        nativeEvent.key   - Chromeï¼Œ Safari, FF
                                                                                        nativeEvent.data  - Safari (IME)
                                                                                      */
      var ch = nativeEvent.char || nativeEvent.key || nativeEvent.data || e.key || e.data;
      if (ch && !formula.test(ch)) {
        e.stopPropagation();
        e.preventDefault();
      }
    }, _this.

    handlePaste = function (e) {var
      formula = _this.props.formula;
      if ((0, _isRegExp2.default)(formula)) {
        var clipboardData = e.clipboardData || window.clipboardData;
        var text = clipboardData.getData('Text');
        if (text && !formula.test(text)) {
          e.stopPropagation();
          e.preventDefault();
        }
      }
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);}(0, _createClass3.default)(Input, [{ key: 'componentDidMount', value: function componentDidMount() {(0, _defineProperties2.default)(this, { value: { get: function get() {return this.input.value;}, set: function set(v) {if (this.props.value === undefined) {this.input.value = v;}} } });} // If user doesn't pass in `value` and only use `defaultValue` as init value or not, then
    // saving it to state
  }, { key: 'render', value: function render() {var _classNames;var _props =
















      this.props,type = _props.type,size = _props.size,preIcon = _props.preIcon,preText = _props.preText,postIcon = _props.postIcon,postText = _props.postText,errored = _props.errored,disabled = _props.disabled,icon = _props.icon,className = _props.className,style = _props.style,PreComponent = _props.PreComponent,PostComponent = _props.PostComponent,ariaLabel = _props.ariaLabel,rest = (0, _objectWithoutProperties3.default)(_props, ['type', 'size', 'preIcon', 'preText', 'postIcon', 'postText', 'errored', 'disabled', 'icon', 'className', 'style', 'PreComponent', 'PostComponent', 'ariaLabel']);
      var inputClassName = (0, _classnames2.default)((_classNames = {

        'input-group': true }, (0, _defineProperty3.default)(_classNames, 'input-group--' +
      size, true), (0, _defineProperty3.default)(_classNames,
      'input-group--icon', icon), (0, _defineProperty3.default)(_classNames,
      'input-group--error', errored), (0, _defineProperty3.default)(_classNames,
      'input-group--disabled', disabled), _classNames),

      className);


      return (
        _react2.default.createElement('div', { className: inputClassName, style: style },
          PreComponent && _react2.default.createElement(PreComponent, { className: 'input-group__item' }),
          (preIcon || preText) &&
          _react2.default.createElement('span', { className: 'input-group__item' },
            preIcon && _react2.default.createElement('i', { className: 'icon ' + preIcon }),
            preText),


          _react2.default.createElement('input', (0, _extends3.default)({
            // the event handler for `onChange` not firing properly in React 15.2.0 + IE11
            // when paste text into textarea: https://github.com/facebook/react/issues/7211
            // will be removed when react fixed this issue
            onInput: this.handleChange },
          (0, _omit2.default)(rest, ['inputRef', 'formula', 'onValueChange', 'onEnter', 'onLeave']), {
            type: type,
            disabled: disabled,
            className: 'input input-group__field input__field',
            ref: this.setWrappedComponentInstance,
            'aria-label': ariaLabel,
            onBlur: this.handleBlur,
            onChange: this.handleChange,
            onKeyPress: this.handleKeyPress,
            onPaste: this.handlePaste })),

          (postIcon || postText) &&
          _react2.default.createElement('span', { className: 'input-group__item' },
            postIcon && _react2.default.createElement('i', { className: 'icon ' + postIcon }),
            postText),


          PostComponent && _react2.default.createElement(PostComponent, { className: 'input-group__item' })));


    } }]);return Input;}(_react.PureComponent);Input.displayName = 'Input';Input.propTypes = InputPropTypes;Input.defaultProps = InputProps;exports.default =


Input;module.exports = exports['default'];