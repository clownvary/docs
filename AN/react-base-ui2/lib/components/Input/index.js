'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.Input = exports.InputField = undefined;var _Input = require('./Input');var _Input2 = _interopRequireDefault(_Input);
var _validation = require('../../services/validation');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var InputField = (0, _validation.createField)(_Input2.default);exports.


InputField = InputField;exports.
Input = _Input2.default;exports.default =


_Input2.default;