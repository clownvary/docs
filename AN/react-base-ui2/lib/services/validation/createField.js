'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _isString = require('lodash/isString');var _isString2 = _interopRequireDefault(_isString);
var _isArray = require('lodash/isArray');var _isArray2 = _interopRequireDefault(_isArray);
var _uniq = require('lodash/uniq');var _uniq2 = _interopRequireDefault(_uniq);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);
var _concat = require('lodash/concat');var _concat2 = _interopRequireDefault(_concat);
var _FieldLayout = require('./consts/FieldLayout');var FieldLayout = _interopRequireWildcard(_FieldLayout);
var _ErrorAlignment = require('./consts/ErrorAlignment');var ErrorAlignment = _interopRequireWildcard(_ErrorAlignment);
var _consts = require('../../consts');
var _Tooltip = require('../../components/Tooltip');var _Tooltip2 = _interopRequireDefault(_Tooltip);
var _buildErrorMessage = require('./utils/buildErrorMessage');var _buildErrorMessage2 = _interopRequireDefault(_buildErrorMessage);
var _Validation = require('./Validation');var _Validation2 = _interopRequireDefault(_Validation);
var _ErrorList = require('./components/ErrorList');var _ErrorList2 = _interopRequireDefault(_ErrorList);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}


var createField = function createField(WrappedComponent) {return function (_PureComponent) {(0, _inherits3.default)(_class2, _PureComponent);

    function _class2(props) {(0, _classCallCheck3.default)(this, _class2);var _this = (0, _possibleConstructorReturn3.default)(this, (_class2.__proto__ || (0, _getPrototypeOf2.default)(_class2)).call(this,
      props));_this.



























      onValidate = function (value) {var _this$props =



        _this.props,_this$props$fieldName = _this$props.fieldName,fieldName = _this$props$fieldName === undefined ? 'Entry' : _this$props$fieldName,onValidated = _this$props.onValidated;

        var result = _this.validator.validate(fieldName, value);
        _this.setState({
          validationError: result.message });


        if ((0, _isFunction2.default)(onValidated)) {
          onValidated(result);
        }
      };_this.

      buildTooltipContent = function (rules) {
        if (!rules || rules.length === 0) return '';var

        fieldName = _this.props.fieldName;
        var ruleSet = _this.validator.getRuleSet();
        return (
          _react2.default.createElement('div', { className: 'an-validation-tooltip' },

            ruleSet.map(function (rule, i) {return (
                _react2.default.createElement('div', { key: i },
                  _react2.default.createElement('span', null, (0, _buildErrorMessage2.default)(rule.message, fieldName, rule.param))));})));





      };_this.

      onEnter = function (target) {var _this$props$showValid =


        _this.props.showValidationTip,showValidationTip = _this$props$showValid === undefined ? false : _this$props$showValid;

        if (showValidationTip) {
          var options = {
            content: _this.buildTooltipContent(),
            dockStyle: _consts.Dock.TOP_LEFT,
            effect: _consts.Effect.FADE,
            distance: 4,
            showShadow: true,
            autoClose: 3000 };


          _Tooltip2.default.open(target, options);
        }
      };_this.

      onLeave = function (value) {
        var onLeave = _this.props.onLeave;

        if ((0, _isFunction2.default)(onLeave)) {
          onLeave(value);
        }

        _this.dirty && _this.onValidate(value);
        _this.dirty = false;
        _Tooltip2.default.close();
      };_this.

      onValueChange = function () {
        _this.dirty = true;var
        validationError = _this.state.validationError;
        if (validationError) {
          _this.setState({
            validationError: '' });

        }
      };_this.

      onError = function (messages) {
        var customMessages = [];
        if ((0, _isString2.default)(messages) && messages !== '') {
          customMessages.push(messages);
        } else if ((0, _isArray2.default)(messages)) {
          customMessages = messages;
        }

        _this.setState({
          customMessages: customMessages });

      };_this.state = { validationError: '', customMessages: [] };_this.dirty = false;return _this;}(0, _createClass3.default)(_class2, [{ key: 'componentDidMount', value: function componentDidMount() {var _props = this.props,_props$rules = _props.rules,rules = _props$rules === undefined ? '' : _props$rules,validationMessages = _props.validationMessages;this.validator = _Validation2.default.createValidator(rules, validationMessages);} }, { key: 'componentWillReceiveProps', value: function componentWillReceiveProps(nextProps) {var nextCustomMessages = nextProps.customMessages;var curCustomMessages = this.props.customMessages;if (curCustomMessages !== nextCustomMessages) {this.onError(nextCustomMessages);}} }, { key: 'render', value: function render()

      {var _this2 = this;var _state =
        this.state,_state$validationErro = _state.validationError,validationError = _state$validationErro === undefined ? '' : _state$validationErro,_state$customMessages = _state.customMessages,customMessages = _state$customMessages === undefined ? [] : _state$customMessages;
        var errorMessages = (0, _concat2.default)([], validationError || [], customMessages);var _props2 =






        this.props,_props2$fieldLayout = _props2.fieldLayout,fieldLayout = _props2$fieldLayout === undefined ? FieldLayout.ROW : _props2$fieldLayout,_props2$fieldLabel = _props2.fieldLabel,fieldLabel = _props2$fieldLabel === undefined ? '' : _props2$fieldLabel,_props2$showLabel = _props2.showLabel,showLabel = _props2$showLabel === undefined ? false : _props2$showLabel,_props2$showError = _props2.showError,showError = _props2$showError === undefined ? false : _props2$showError,_props2$errorAlignmen = _props2.errorAlignment,errorAlignment = _props2$errorAlignmen === undefined ? ErrorAlignment.BOTTOM : _props2$errorAlignmen;

        var clsFieldLayout = 'an-field-layout-' + fieldLayout;
        var clsErrorLayout = 'an-error-layout-' + errorAlignment;
        return (
          _react2.default.createElement('div', { className: (0, _classnames2.default)('an-validation-field', clsErrorLayout) },
            _react2.default.createElement('div', { className: (0, _classnames2.default)('an-validation-field__wrap', clsFieldLayout) },
              showLabel && _react2.default.createElement('span', { className: 'an-validation-field__label' }, fieldLabel),
              _react2.default.createElement('div', { className: (0, _classnames2.default)('an-validation-field__content') },
                _react2.default.createElement(WrappedComponent, {
                  ref: function ref(c) {_this2.wrappedComponent = c;},
                  onEnter: function onEnter(_ref) {var target = _ref.target;return _this2.onEnter(target);},
                  onLeave: function onLeave(_ref2) {var value = _ref2.value;return _this2.onLeave(value);},
                  onValueChange: function onValueChange(_ref3) {var value = _ref3.value;return _this2.onValueChange(value);},
                  onError: function onError(_ref4) {var messages = _ref4.messages;return _this2.onError(messages);} }))),



            showError && _react2.default.createElement(_ErrorList2.default, { className: 'an-validation-field__error', messages: (0, _uniq2.default)(errorMessages) })));


      } }]);return _class2;}(_react.PureComponent);};exports.default =


createField;module.exports = exports['default'];