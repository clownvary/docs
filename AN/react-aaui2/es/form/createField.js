import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _extends from 'babel-runtime/helpers/extends';
import React, { createElement } from 'react';

import getDisplayName from '../shared/getDisplayName';
import { FormFieldPropTypes, FormStorePropTypes } from './types';
import { parseRules } from './validation';

var formFields = [];
var propTypes = _extends({}, FormFieldPropTypes);

export var registerFormField = function registerFormField(Field) {
  if (formFields.indexOf(Field) < 0) {
    formFields.push(Field);
  }
};
export var isRegisteredField = function isRegisteredField(type) {
  return formFields.indexOf(type) >= 0;
};

var createField = function createField(FieldLayout) {
  var WrappedField = function WrappedField(_ref, context) {
    var component = _ref.component,
        rest = _objectWithoutProperties(_ref, ['component']);

    var name = rest.name;
    var store = context.aauiFormStore;

    var fieldState = store.getState()[name] || {};

    // Consider `props.rules` first then field state
    var rules = rest.rules || fieldState.rules || '';
    var required = Object.prototype.hasOwnProperty.call(rest, 'required') ? rest.required : rules.indexOf('required') !== -1;

    var element = createElement(component, _extends({}, rest, fieldState, {
      rules: rules,
      required: required
    }));

    var parsedRules = parseRules(rules);
    var l10nMessageValues = parsedRules.reduce(function (r, rule) {
      var _extends2;

      return _extends({}, r, (_extends2 = {}, _extends2[rule.name] = rule.param, _extends2));
    }, {});

    if (FieldLayout) {
      return React.createElement(
        FieldLayout,
        _extends({
          value: fieldState.value,
          errMsg: fieldState.errMsg,
          required: required,
          l10nMessageValues: l10nMessageValues
        }, rest),
        element
      );
    }

    return element;
  };

  WrappedField.propTypes = propTypes;
  WrappedField.contextTypes = FormStorePropTypes;
  WrappedField.displayName = 'WrappedField(' + getDisplayName(WrappedField) + ')';

  return WrappedField;
};

export default createField;