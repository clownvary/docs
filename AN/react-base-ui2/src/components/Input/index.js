import Input from './Input';
import { createField } from '../../services/validation';

const InputField = createField(Input);

export {
  InputField,
  Input
};

export default Input;
