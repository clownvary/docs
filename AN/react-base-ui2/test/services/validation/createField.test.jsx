import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Input from 'src/components/Input';
import createField from 'src/services/validation/createField';


describe('services/validation/createField.jsx', () => {
  const setup = (props = {}, WrappedComponent = Input) => {
    const Field = createField(WrappedComponent);
    return mount(<Field {...props} />);
  };

  test('rendering', () => {
    const wrapper = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setProps({ customMessages: 'test message' });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setProps({ customMessages: '' });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setProps({ customMessages: ['array messages'] });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setProps({ showLabel: true, showError: true });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper.setState({ validationError: undefined, customMessages: undefined });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('handle events', () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    const inputComponent = wrapper.find(Input);
    const input = inputComponent.find('input');
    const onLeave = jest.fn();
    const onValidated = jest.fn();

    jest.spyOn(instance, 'buildTooltipContent');
    jest.spyOn(instance, 'onValidate');

    inputComponent.prop('onEnter')({ target: input });
    expect(instance.buildTooltipContent).not.toBeCalled();

    wrapper.setProps({ showValidationTip: true });
    inputComponent.prop('onEnter')({ target: input });
    expect(instance.buildTooltipContent).toHaveBeenCalledTimes(1);

    instance.dirty = true;
    inputComponent.prop('onLeave')({ value: 'test' });
    expect(instance.onValidate).toHaveBeenCalledTimes(1);
    expect(onLeave).not.toBeCalled();

    wrapper.setProps({ onLeave, onValidated });
    instance.dirty = true;
    inputComponent.prop('onLeave')({ value: 'test' });
    expect(onLeave).toHaveBeenCalledTimes(1);
    expect(onValidated).toHaveBeenCalledTimes(1);

    inputComponent.prop('onValueChange')({ value: 'test' });
    expect(wrapper.state('validationError')).toBe('');

    wrapper.setState({ validationError: 'validation error' });
    inputComponent.prop('onValueChange')({ value: 'test' });
    expect(wrapper.state('validationError')).toBe('');

    inputComponent.prop('onError')({ messages: 'test messages' });
    expect(wrapper.state('customMessages')).toEqual(['test messages']);

    const wrapper2 = setup({ rules: 'required' });
    const instance2 = wrapper2.instance();
    expect(toJSON(instance2.buildTooltipContent(['required']))).toMatchSnapshot();
  });
});
