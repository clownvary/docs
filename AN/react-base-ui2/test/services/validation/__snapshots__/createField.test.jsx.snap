// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`services/validation/createField.jsx handle events 1`] = `undefined`;

exports[`services/validation/createField.jsx rendering 1`] = `
<_class2>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
  </div>
</_class2>
`;

exports[`services/validation/createField.jsx rendering 2`] = `
<_class2
  customMessages="test message"
>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
  </div>
</_class2>
`;

exports[`services/validation/createField.jsx rendering 3`] = `
<_class2
  customMessages=""
>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
  </div>
</_class2>
`;

exports[`services/validation/createField.jsx rendering 4`] = `
<_class2
  customMessages={
    Array [
      "array messages",
    ]
  }
>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
  </div>
</_class2>
`;

exports[`services/validation/createField.jsx rendering 5`] = `
<_class2
  customMessages={
    Array [
      "array messages",
    ]
  }
  showError={true}
  showLabel={true}
>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <span
        className="an-validation-field__label"
      />
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
    <ErrorList
      className="an-validation-field__error"
      messages={
        Array [
          "array messages",
        ]
      }
    >
      <div
        className="an-error-list an-validation-field__error"
      >
        <div
          className="an-error"
          key="0"
        >
          <span
            className="icon aaui-alert-error-icon icon-times-circle"
          />
          <span>
            array messages
          </span>
        </div>
      </div>
    </ErrorList>
  </div>
</_class2>
`;

exports[`services/validation/createField.jsx rendering 6`] = `
<_class2
  customMessages={
    Array [
      "array messages",
    ]
  }
  showError={true}
  showLabel={true}
>
  <div
    className="an-validation-field an-error-layout-bottom"
  >
    <div
      className="an-validation-field__wrap an-field-layout-row"
    >
      <span
        className="an-validation-field__label"
      />
      <div
        className="an-validation-field__content"
      >
        <Input
          ariaLabel="Input box"
          icon={false}
          inputRef={[Function]}
          onChange={[Function]}
          onEnter={[Function]}
          onError={[Function]}
          onLeave={[Function]}
          onValueChange={[Function]}
          size="md"
          type="text"
        >
          <div
            className="input-group input-group--md"
          >
            <input
              aria-label="Input box"
              className="input input-group__field input__field"
              onBlur={[Function]}
              onChange={[Function]}
              onError={[Function]}
              onInput={[Function]}
              onKeyPress={[Function]}
              onPaste={[Function]}
              type="text"
            />
          </div>
        </Input>
      </div>
    </div>
    <ErrorList
      className="an-validation-field__error"
      messages={Array []}
    >
      <div
        className="an-error-list an-validation-field__error"
      />
    </ErrorList>
  </div>
</_class2>
`;
